src/main/webapp/app/overview/course-score-calculation.service.ts

```diff
@@ -150,7 +150,7 @@ export class CourseScoreCalculationService {
             });
 
             const gracePeriodInSeconds = 10;
-            if (dueDate === null || dueDate.add(gracePeriodInSeconds, 'seconds') >= resultsArray[0].completionDate!) {
+            if (!!!dueDate || dueDate.add(gracePeriodInSeconds, 'seconds') >= resultsArray[0].completionDate!) {
```

Comment:
Would this be the same thing?
```suggestion
            if (dueDate == null || dueDate.add(gracePeriodInSeconds, 'seconds') >= resultsArray[0].completionDate!) {
```

Reply:
This also checks for undefined

Reply:
@jpbernius This should work as well. `==` checks for `null` and `undefined`.

Reply:
I feel like `!!!` is less clear than `== null`.

Reply:
@sjagla27 Note `==` vs `===`

Reply:
Okay.